cvat-ui/src/actions/annotation-actions.ts

```diff
@@ -1608,3 +1611,84 @@ export function restoreFrameAsync(frame: number): ThunkAction {
         }
     };
 }
+
+export function updateEditedStateAsync(
+    shapeType: ShapeType | null,
+    editedStateInstance: ObjectState | null,
+): ThunkAction {
+    return async (dispatch: ThunkDispatch, getState): Promise<void> => {
+        if (editedStateInstance) {
+            const state = getState();
+            const { instance: canvas } = state.annotation.canvas;
+            if (canvas) {
+                (canvas as Canvas).configure({
+                    hideEditedObject: editedStateInstance.hidden,
+                });
+            }
+
+            dispatch({
+                type: AnnotationActionTypes.HIDE_EDITED_STATE,
+                payload: {
+                    hide: editedStateInstance.hidden,
+                },
+            });
+        }
```

Comment:
I think we have three actions `changeHideEditedStateAsync`, `updateEditedStateAsync`, `resetEditedStateAsync` doing the same thing (updating the same key in store).

Probably we may combine them to reduce the code duplication. Let it accept `shapeType`, `objectState` and `isHidden`. And use these arguments to update store and objectState properly

Reply:
Considering my general review comment it should only accept `objectState: ObjectState | null`

Reply:
Now we have two actions: `updateEditedStateAsync` for manipulation of edited state object, another - `changeHideActiveObjectAsync` to change the hide flag. I put the hide flag in `canvas` part of the store.